### PR TITLE
Feat(eos_cli_config_gen): Adding 'ip igmp host-proxy' interface support

### LINKED ISSUE #3633: Feat(eos_cli_config_gen): Add 'ip igmp host-proxy' interface support
### Enhancement summary

Add support to configure IP IGMP Host-Proxy feature:

````code
vEOS-4-30-5M(config-if-Et2)#ip igmp host-proxy ?
  A.B.C.D          Group Address
  access-list      Non-standard Access List name
  report-interval  Time interval between unsolicited reports
  version          IGMP version on IGMP host-proxy interface
  <cr>
````

### Which component of AVD is impacted

eos_cli_config_gen

### Enable

````yaml
ethernet_interfaces:
  - name: Ethernet1
    ip_igmp_host_proxy:
      enabled: true
````
should render:

````code
interface Ethernet1
   ip igmp host-proxy
````

### Group Filter

````yaml
ethernet_interfaces:
  - name: Ethernet1
    ip_igmp_host_proxy:
      groups:
         - group: 239.0.0.1
         - group: 239.0.0.2
           exclude:
             - source: 10.0.2.1
         - group: 239.0.0.3
           include:
             - source: 10.0.3.1
         - group: 239.0.0.4
           exclude:
             - source: 10.0.4.1
             - source: 10.0.4.2
           include:
             - source: 10.0.4.3
             - source: 10.0.4.4
````
should render:

````code
interface Ethernet1
   ip igmp host-proxy 239.0.0.2 exclude 10.0.2.1
   ip igmp host-proxy 239.0.0.3 include 10.0.3.1
   ip igmp host-proxy 239.0.0.1
   ip igmp host-proxy 239.0.0.4 include 10.0.4.3
   ip igmp host-proxy 239.0.0.4 include 10.0.4.4
   ip igmp host-proxy 239.0.0.4 exclude 10.0.4.1
   ip igmp host-proxy 239.0.0.4 exclude 10.0.4.2
````
````

### Report Interval

````yaml
ethernet_interfaces:
  - name: Ethernet1
    ip_igmp_host_proxy:
      report_interval: 2
````
should render:

````code
interface Ethernet1
   ip igmp host-proxy report-interval 2
````

### Access Lists

````yaml
ethernet_interfaces:
  - name: Ethernet1
    ip_igmp_host_proxy:
      access_lists:
        - name: ACL1
        - name: ACL2
````
should render:

````code
interface Ethernet1
   ip igmp host-proxy access-list ACL1
   ip igmp host-proxy access-list ACL2
````

### Version

````yaml
ethernet_interfaces:
  - name: Ethernet1
    ip_igmp_host_proxy:
      version: 2
````
should render:

````code
interface Ethernet1
   ip igmp host-proxy version 2
````

### Describe the solution you would like

data_model on ethernet, port-channel and vlan interfaces:

````yaml
ip_igmp_host_proxy:
  enabled: Boolean
  access_lists:
    - name: String  # ACL name
  groups:
    - group: String  # IPv4 Multicast Group; required
      exclude:
        - source: String  # IPv4 source address
      include:
        - source: String  # IPv4 source address
  report_interval: Integer  # allowed values: <1-31744>; default: 1
  version: Integer  # allowed_values: <1-3>; default: 3
````

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary

Adding 'ip igmp host-proxy' interface support

## Related Issue(s)

Fixes #3633  , #3566 

## Component(s) name

`arista.avd.eos_cli_config_gen`

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml [hunk 1]
```diff
@@ -758,3 +758,66 @@ keys:
...
+                - str
+              min: 1
+              max: 31744
+              default: 1
+              description: Time interval between unsolicited reports.
+            access_lists:
+              type: list
+              primary_key: name
+              convert_types:
+                - dict
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml]: No need for this in new schemas.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml [hunk 2]
```diff
@@ -758,3 +758,66 @@ keys:
         eos_cli:
           type: str
           description: Multiline EOS CLI rendered directly on the port-channel interface in the final EOS configuration
+        ip_igmp_host_proxy:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml:755]: Please move this above the `peer` metadata key.

Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml:755]: Done!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml (hunk 1 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml]: Done!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml [hunk 3]
```diff
@@ -731,6 +731,69 @@ keys:
...
+                - str
+              min: 1
+              max: 31744
+              default: 1
+              description: Time interval between unsolicited reports.
+            access_lists:
+              type: list
+              primary_key: name
+              # convert_types:
+              #   - dict
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml [hunk 4]
```diff
@@ -731,6 +731,69 @@ keys:
...
+                      keys:
+                        source:
+                          type: str
+            report_interval:
+              type: int
+              convert_types:
+                - str
+              min: 1
+              max: 31744
+              default: 1
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml]: We should not have defaults in eos_cli_config_gen.
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml [hunk 5]
```diff
@@ -731,6 +731,69 @@ keys:
...
+                    items:
+                      type: dict
+                      keys:
+                        source:
+                          type: str
+                  include:
+                    type: list
+                    primary_key: source
+                    # convert_types:
+                    #   - dict
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml [hunk 6]
```diff
@@ -731,6 +731,69 @@ keys:
...
+              items:
+                type: dict
+                keys:
+                  group:
+                    type: str
+                  exclude:
+                    type: list
+                    primary_key: source
+                    # convert_types:
+                    #   - dict
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml [hunk 7]
```diff
@@ -731,6 +731,69 @@ keys:
...
+        ip_igmp_host_proxy:
+          type: dict
+          keys:
+            enabled:
+              type: bool
+            groups:
+              type: list
+              primary_key: group
+              # convert_types:
+              #   - dict
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml [hunk 8]
```diff
@@ -731,6 +731,69 @@ keys:
...
+                keys:
+                  name:
+                    type: str
+            version:
+              type: int
+              convert_types:
+                - str
+              min: 1
+              max: 3
+              default: 3
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml]:
```suggestion
```

**Review by ClausHolbechArista — COMMENTED** [review]
Please adjust all schema fragments accordingly.

**Comment by ClausHolbechArista** [comment]
@philippebureau @kornoa You have requested this feature, so I hope you can help us test this PR.
You can install with galaxy directly from the PR like this:
```sh
ansible-galaxy collection install git+https://github.com/vibhu-gslab/ansible-avd.git#/ansible_collections/arista/avd/,igmp_host_proxy --force
```

**Comment by kornoa** [comment]
@ClausHolbechArista:
I've checked the adjusted molecule test in the PR. This looks good. I don't need more local testing. Relevant testing should always be included in the code. ;-)
Besides functionality: I would recommend to outsource repetitive Jinja2 code from various interface type templates. Maintenance on duplicated code is annoying.

**Comment by ClausHolbechArista** [comment]
@kornoa 
`Besides functionality: I would recommend to outsource repetitive Jinja2 code from various interface type templates. Maintenance on duplicated code is annoying.`
Ack, but Jinja is suboptimal for this. We try to balance the ease of contributions by non-programmers vs the efficiency of the code.
We will evaluate your suggestion. Thanks.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/vlan-interfaces.md [hunk 9]
```diff
@@ -230,6 +242,26 @@
         ip_verify_unicast_source_reachable_via: <str; "any" | "rx">
         ip_igmp: <bool>
         ip_igmp_version: <int; 1-3>
+        ip_igmp_host_proxy:
+          enabled: <bool>
+
+          # Group Address.
+          groups:
+            - group: <str; required; unique>
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/vlan-interfaces.md:255]: Group address should be a valid multicast address

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml (hunk 3 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml]: Done!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml (hunk 4 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml]: Done!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml (hunk 5 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml]: Done!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml (hunk 6 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml]: Done!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml (hunk 7 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml]: Done!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml (hunk 8 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/port_channel_interfaces.schema.yml]: Done!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/vlan-interfaces.md (hunk 9 above)
Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/tables/vlan-interfaces.md:255]: Done!

**Review by MaheshGSLAB — APPROVED** [review]

**Review by philippebureau — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/vlan_interfaces.schema.yml [hunk 10]
```diff
@@ -89,6 +89,59 @@ keys:
...
+            groups:
+              type: list
+              primary_key: group
+              items:
+                type: dict
+                keys:
+                  group:
+                    type: str
+                    description: Multicast Address.
+                  exclude:
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/vlan_interfaces.schema.yml:106]: Add you cannot have the same IP both in include and exclude in a description

**Review by gmuloc — CHANGES_REQUESTED** [review]
one small request for the description.

Couple of points I checked:
* ordering of groups and include / exclude lists and ACLs.
    * for ACL it is quite messy as it is implemented as a dictionary and it is quite not deterministic so users would have to adjust the inputs if they want to match - this reflects neither the order of insertion not an alphabetical order:
        ```
        interface Ethernet1
           description ATT_Colt55
           no switchport
           ip igmp host-proxy
           ip igmp host-proxy access-list ACL1
           ip igmp host-proxy access-list ACL2
           ip igmp host-proxy access-list BLAH3
           ip igmp host-proxy access-list BLEH
           ip igmp host-proxy access-list ACL3
           ip igmp host-proxy access-list BLAH2
           ip igmp host-proxy access-list BLAH
        ```
    * for groups, it seems to follow the order of insertion so what we have is good
    * for includ and exclude, sorting could be envisioned as it seems to follow mostly this but I am seeing unconsistent behavior with two entries.
        ```
           ip igmp host-proxy
           ip igmp host-proxy 224.1.1.1 include 1.1.1.1
           ip igmp host-proxy 224.1.1.1 include 1.1.1.9
           ip igmp host-proxy 224.1.1.1 include 1.1.1.10
           ip igmp host-proxy 224.1.1.1 include 1.1.1.0.  <<<<<<<<< always last
       ```
       and
       ```
         ip igmp host-proxy 224.1.1.1 include 1.1.1.1
         ip igmp host-proxy 224.1.1.1 include 2.2.2.2
         ip igmp host-proxy 224.1.1.1 include 5.5.5.5
         ip igmp host-proxy 224.1.1.1 exclude 1.1.1.4
         ip igmp host-proxy 224.1.1.1 exclude 1.1.1.9
         ip igmp host-proxy 224.1.1.1 exclude 1.1.1.2.  <<< 
       ```

* For now no documentation added to doc templates. let's see if it is a request in the future but otherwise let's merge like this.

**Review by gmuloc — APPROVED** [review]

**Pull request merged**